src/core/dependencies.jl

```diff
@@ -1,3 +1,30 @@
+using JuliaVariables
+using MLStyle
+import MacroTools
+
+
+
+unwrap_scoped(ex) = @match ex begin
+    Expr(:scoped, _, a) => unwrap_scoped(a)
+    Expr(head, args...) => Expr(head, map(unwrap_scoped, args)...)
+    a => a
+end
+
+globals(s::Symbol) = [s]
```

Comment:
This way is correct but it maybe faster if we use side-effect.
I'm not sure if this is performance-critical.
Others look fine to me.

Reply:
Thanks Taine. For now it's not, but that may change as we extend things. Let's go with this for now, and we can come back to it if it becomes a bottleneck.